STF 1.1 documentation updates (round 1) (#136)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/con_manifest-overrides.adoc

The Service Telemetry Operator watches for a `ServiceTelemetry` manifest to load into {OpenShift} ({OpenShiftShort}). The Operator then creates other objects in memory, which results in the dependent Operators creating the workloads they are responsible for managing.

[WARNING]
====
When you override the manifest, you must provide the entire manifest contents, including object names or namespaces. There is no dynamic parameter substitution when you override a manifest.

> Using manifest overrides is not recommended and should be used as a last resort short circuit.

Reviewer: ```suggestion
Use manifest overrides only as a last resort short circuit.
```

We don't like to recommend things in general. We like to say, do this or do not do this. 

I suggest here we simply say "Use manifest overrides only as a last resort short circuit."

====

To override a manifest successfully with {Project} ({ProjectShort}), deploy a default environment using the core options only. For more information about the core options, see xref:creating-a-servicetelemetry-object-in-openshift[]. When you deploy {ProjectShort}, use the `oc get` command to retrieve the default deployed manifest. When you use a manifest that was originally generated by Service Telemetry Operator, the manifest is compatible with the other objects that are managed by the Operators.

> For example, when the `backends.metrics.prometheus.enabled: true` parameter is configured in the `ServiceTelemetry` manifest, the Service Telemetry Operator requests components for metrics retrieval and storage using the default manifests. In some cases, you might want to override the default manifest. For more information, see xref:manifest-override-parameters_advanced-features[].

Reviewer: ```suggestion
For example, when the `backends` parameter for Prometheus metrics is enabled in the `ServiceTelemetry` manifest, the Service Telemetry Operator requests components for metrics retrieval and storage using the default manifests. In some cases, you might want to override the default manifest. For more information, see xref:manifest-override-parameters_advanced-features[].
```

I really don't like this, I know it makes things simpler for the docs, but it doesn't match what the user sees in the example configurations.

Also I know there is a note somewhere saying for the purpose of the docs we will use this format, however, each module should be standalone. You really would have to paste that note into every module in which we use this parameter description.

Author: Talked with Joanne and we'll investigate what we should do here. I can't really apply this change because `backends` is for both `metrics` and `events` configuration so changing this to just `backends` makes the test confusing to me. We're checking with documentation team to figure out the proper nested notation.

Author: Actually re-reading this, it's probably fine. I'll come back after reviewing the other changes.
